cmd/juju/application/removerelation_test.go

```diff
@@ -1,78 +1,94 @@
-	s.RepoSuite.SetUpTest(c)
-	s.CmdBlockHelper = testing.NewCmdBlockHelper(s.APIState)
-	c.Assert(s.CmdBlockHelper, gc.NotNil)
-	s.AddCleanup(func(*gc.C) { s.CmdBlockHelper.Close() })
+	s.IsolationSuite.SetUpTest(c)
+	s.mockAPI = &mockRemoveAPI{
+		removeRelationFunc: func(endpoints ...string) error {
+			return nil
+		},
+	}
 }
 
 var _ = gc.Suite(&RemoveRelationSuite{})
 
-func runRemoveRelation(c *gc.C, args ...string) error {
-	_, err := testing.RunCommand(c, NewRemoveRelationCommand(), args...)
+func (s *RemoveRelationSuite) runRemoveRelation(c *gc.C, args ...string) error {
+	_, err := coretesting.RunCommand(c, NewRemoveRelationCommandForTest(s.mockAPI), args...)
 	return err
 }
 
-func (s *RemoveRelationSuite) setupRelationForRemove(c *gc.C) {
-	ch := testcharms.Repo.CharmArchivePath(s.CharmsPath, "riak")
-	err := runDeploy(c, ch, "riak", "--series", "quantal")
-	c.Assert(err, jc.ErrorIsNil)
-	ch = testcharms.Repo.CharmArchivePath(s.CharmsPath, "logging")
-	err = runDeploy(c, ch, "logging", "--series", "quantal")
+func (s *RemoveRelationSuite) TestRemoveRelationNotEnoughArguments(c *gc.C) {
+	// No arguments
+	err := s.runRemoveRelation(c)
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty argument
+	err = s.runRemoveRelation(c, "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 2 empty arguments
+	err = s.runRemoveRelation(c, "", "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty and 1 non-empty arguments
+	err = s.runRemoveRelation(c, "application1", "")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+
+	// 1 empty and 1 non-empty arguments
+	err = s.runRemoveRelation(c, "", "application2")
+	c.Assert(err, gc.ErrorMatches, "a relation must involve two applications")
+}
+
+func (s *RemoveRelationSuite) TestRemoveRelationSuccess(c *gc.C) {
+	err := s.runRemoveRelation(c, "application1", "application2")
 	c.Assert(err, jc.ErrorIsNil)
-	runAddRelation(c, "riak", "logging")
 }
 
-func (s *RemoveRelationSuite) TestRemoveRelation(c *gc.C) {
-	s.setupRelationForRemove(c)
+func (s *RemoveRelationSuite) TestRemoveRelationFail(c *gc.C) {
+	msg := "fail remove relation"
+	s.mockAPI = &mockRemoveAPI{
+		removeRelationFunc: func(endpoints ...string) error {
+			return errors.New(msg)
+		},
+	}
+	err := s.runRemoveRelation(c, "application1", "application2")
+	c.Assert(err, gc.ErrorMatches, msg)
+}
 
-	// Destroy a relation that exists.
-	err := runRemoveRelation(c, "logging", "riak")
-	c.Assert(err, jc.ErrorIsNil)
+func (s *RemoveRelationSuite) TestRemoveRelationBlocked(c *gc.C) {
+	// Block operation
+	s.mockAPI.removeRelationFunc = func(endpoints ...string) error {
+		return common.OperationBlockedError("TestRemoveRelationBlocked")
+	}
+	err := s.runRemoveRelation(c, "application1", "application2")
+	coretesting.AssertOperationWasBlocked(c, err, ".*TestRemoveRelationBlocked.*")
```

Comment:
We do not return anything user-friendly to console. We log blocked operations. I wonder if it's right though... If it's not, I am tempted to address it in a separate PR as it will involve change in behavior of all blocked commands.

I have added a check to ensure that the log gets correct message with currently existing behavior.